### DISCUSSION TITLE
Is there any way to install packages without layers?

### DISCUSSION THREAD

**Discussion by BrandonShega** [syl20bnr/spacemacs#15072] [Q&A] [answered]
I'm trying to set up a literate config for Spacemacs and I have everything working except for my private layers that I have packages set up in.  Going through the Spacemacs documentation it looks like the only way to install packages is by using layers, having them in the `packages.el` file, putting them in a list and then having an init function for each one.  Is it possible to use `package.el` and just having `ensure: t` on them to install them? Or is there a way to implement custom layers somehow with a literate config?

**Comment by Novite5** [comment]
You can just install packages by adding them to `dotspacemacs-additional-packages` section in `.spacemacs`.

**Comment by BrandonShega** [comment]
Do I still have to have an init line for each one in a layer? Or can I just use `use-package` like normal?

**Comment by Novite5** [comment]
In most instances, both are unnecessary.

**Comment by BrandonShega** [comment]
It looks I probably have to have 2 org files, one for `dotspacemacs/init` and one for `dotspacemacs/user-load` or I guess maybe I can just keep everything in the init portion that comes from Spacemacs and make the load portion my actual config?

**Comment by Novite5** [comment]
there is a ˋuser-configˋ section in .spacemacs, you can put configuration for packages in it.

**Comment by emacs18** [comment]
I would set `dotspacemacs-install-packages` variable in `~/.spacemacs` to `'used-but-keep-unused`.  This prevents spacemacs from removing packages unknown to spacemacs on startup.

Next just use `use-package` as you would have done without spacemacs to install packages.  This is how use my personal setup code that I've been using prior to spacemacs, so that I can start emacs with or without spacemacs, and with or withou my personal setup.

**Comment by pataquets** [comment, marked as answer]
For reference, this the official documentation:
https://develop.spacemacs.org/doc/FAQ.html#install-a-package-not-provided-by-a-layer
